### PR TITLE
Initial bugsnag integration

### PR DESCRIPTION
**Pull request by belemaire**
This is the initial integration of [bugsnag](https://www.bugsnag.com/) in electrode native.

When regenerating a cauldron container, or performing a code-push release, if a matching `bugsnag` configuration is found in the cauldron for target descriptor, electrode native will then upload the JS bundle source map -along with associated sources- to the bugsnag server, using [ern-bugsnag-sourcemaps](https://github.com/electrode-io/ern-bugsnag-sourcemaps) library.

It will also generate a unique `codeBundleId` that will be provided in the request to upload the source map. This `codeBundleId` is used by bugsnag to match a bundle to a source map. 
Electrode Native will also set this `codeBundleId` as a process environment variable `ERN_BUGSNAG_CODE_BUNDLE_ID`, prior to composite generation. 
This way, some miniapp in the composite (or through base composite top level code) in charge of global packages (horizontals) initialization, can pick up this bundle id to initialize bugsnag, for example using [babel-plugin-transform-inline-environment-variables](https://www.npmjs.com/package/babel-plugin-transform-inline-environment-variables). 

Refer to [bugsnag documentation](https://docs.bugsnag.com/platforms/react-native/react-native/showing-full-stacktraces/) for more details regarding this `codeBundleId`.

Please note that doing this third party vendor integration directly in electrode native core is not ideal. It was done because of time constraints. Ideally such third party integration should be redistributed in some sort of separate electrode native plugin, as done for container publishers and transformers. We don't have any support -yet- for plugins sitting at the composite/bundling level (only post container generation plugins), but should have soon hopefully. This way we could externalize this bugsnag integration to a dedicated package (same for our in-house integration for our ern source map store).

No documentation update yet. This integration work is mostly meant for internal use only at this point. We will release public documentation later on, once battle tested internally.

### CODE REVIEW

**Comment by belemaire** [comment]
@friederbluemle 
As of today, the only kind of packages that can be 'plugged' in Electrode Native and not baked into core platform modules are Containers transformers and publishers. These optional 'plugins' are invoked by Electrode Native after container project generation is complete, to perform extra custom transformations of the container project and any publication.

For bugsnag (or even for our own publication to ern source map store), as it is implemented at the moment, we would need the ability to plugin a package at JS bundle generation level (with one of its function called just before bundle generation -to patch composite code-, and another invoked just after bundle is generated -to upload source map-). 

In theory it shouldn't be very complex to do so, but don't want to jump on a quick implementation. Want us to think about it more in depth beforehand. For sure we should offer some kind of way to plug-in some custom code at different lifecycle points (pre/post composite gen, pre/post bundle gen, pre/post container gen ...), as it is done right now only for post container generation with publishers / transformers. 

Now there are some questions about the proper design of this solution. Don't want to rush it, because once interface to this approach is released its harder to change (as it can break external implementers) compared to changing things that are internal to our core platform. Given time constraints (pressing internal need for bugsnag) it's best to just keep bugsnag as it is in this PR, as it doesn't cause any issue/complication apart from being a design smell. It will be easy to extract it once we reach a proper architecture for ern "plugins".

**Comment by friederbluemle** [comment]
Thanks for the detailed explanation @belemaire :+1:

I'm fine with getting this merged now. But let's definitely try to address this in a more sustainable and scaleable way very soon. Something like you mentioned with a generic lifecycle hook based plugin architecture. Once we have that in place, we can take out all Bugsnag related code and attach it in a more loosely coupled fashion.

File: ern-orchestrator/src/constants.ts [hunk 1]
```diff
@@ -76,4 +76,9 @@ export const availableUserConfigKeys = [
     name: 'binaryStoreProxy',
     values: ['string'],
   },
+  {
+    desc: 'HTTP/HTTPS proxy to use to connect to bugsnag',
```
[the remark this answers is not in the document]
Comment (friederbluemle) [reply ern-orchestrator/src/constants.ts]: Another minor request: Looks like the official spelling based on their [website](https://www.bugsnag.com/) is _Bugsnag_ (with a capital B) - Would be good to keep that spelling in all regular text.

File: ern-core/package.json [hunk 2]
```diff
@@ -60,6 +60,7 @@
     "@yarnpkg/lockfile": "^1.1.0",
     "archiver": "^2.1.1",
     "bluebird": "^3.5.1",
+    "ern-bugsnag-sourcemaps": "^1.0.0",
```
[the remark this answers is not in the document]
Comment (belemaire) [reply ern-core/package.json]: Good guess ;)

File: ern-orchestrator/src/constants.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (belemaire) [reply ern-orchestrator/src/constants.ts]: Updated string.
For the proxy we currently have three different proxy config that can be configured. One for binary store, one for source map store, and now one for bugsnag.
Idea is that some setup could configure different proxies for these stores (or no proxy at all if needed). Could revisit this later on.

**Pull request merged**